webapp/src/components/views/BackendProbe.tsx

```diff
@@ -11,20 +13,53 @@ interface IData {
 
 export const BackendProbe: FC<IData> = ({ uri, onBackendFound }) => {
     const classes = useSharedClasses();
+    const { isMigrating } = useAppSelector((state: RootState) => state.app);
+    const healthUrl = new URL('healthz', uri);
+    const migrationUrl = new URL('migrationstatus', uri);
+
+    console.log(`# ${isMigrating} (PROBE)`); // $$$
+
     useEffect(() => {
         const timer = setInterval(() => {
-            const requestUrl = new URL('healthz', uri);
-            const fetchAsync = async () => {
-                const result = await fetch(requestUrl);
+            const fetchHealthAsync = async () => {
+                const result = await fetch(healthUrl);
+
+                if (isMigrating) {
+                    console.log(`# MIGRATING`); // $$$
+                    return;
+                }
 
                 if (result.ok) {
+                    console.log(`# HEALTH DONE`); // $$$
+                    onBackendFound();
+                }
+
+                console.log(`# HEALTH NEXT`); // $$$
+            };
+
+            const fetchMigrationAsync = async () => {
+                const result = await fetch(migrationUrl);
+
+                if (!result.ok) {
+                    return;
+                }
+
+                const text = "test" + 1; // $$$
+
+                if (text === 'None') {
                     onBackendFound();
                 }
             };
 
-            fetchAsync().catch(() => {
+            fetchHealthAsync().catch(() => {
```

Comment:
Optimization since health will always be verified before migration check (optionally, you can put this in an else statement below)

```suggestion
            if (!isMigrating) {
                   fetchHealthAsync().catch(() => {
```

then you can also remove the isMigrating check in lines 27-30